### ISSUE TITLE
Updating also unuses dependencies

### ISSUE THREAD

**Issue by GeorgHoffmeyer** [littlerobots/version-catalog-update-plugin#65]
We use the version catalog, to share versions across multiple projects. For that we have a dependancy-managment project, which just publishs the toml file to our artifact repository.

I now want to use the version-catalog-update plugin to update the toml file. To do so, I must define all versions in the catalog as a dependencie in my buidl.gradle.

It would be great if I could just use the toml file and update everything that is defined in that file, no matter if it is used as a dependency or not.

**Comment by hvisser** [comment]
Yeah that makes sense, however it's more or less backwards to what this plugin is currently doing (the versions plugin is actually doing all of the dependency resolving, based on what is used in the project). Ideally we'd still want to go through that plugin so that any component selection rules etc are the same, otherwise the plugin would need its own mechanism for detecting updates and rules around how to select those, duplicating the functionality of the versions plugn.

**Comment by hvisser** [comment]
I think you might be able to do this today, if you use the versions catalog api to add all the libraries in your toml file(s) as a dependency, something like this:

```
val versionCatalog = extensions.getByType<VersionCatalogsExtension>().named("libs")

dependencies {
    versionCatalog.libraryAliases.forEach {alias ->
        versionCatalog.findLibrary(alias).ifPresent {
            implementation(it)
        }
    }
}
```

You'd need to do something similar for plugins (by their plugin id) which might be trickier. But once you have those added the update should work as normal.

**Comment by hvisser** [comment]
Did the workaround I mentioned work for you @GeorgHoffmeyer? If it does I think the plugin does not need more additional support for now.

**Comment by GeorgHoffmeyer** [comment]
Yes it worked for the dependencies. For plugins I have no solution yet.

**Comment by hvisser** [comment]
I think the easiest way for handling plugins would be add them to your build script in the plugin block with `apply false`. I don't know of any Gradle APIs that can add a plugin by its plugin id like the plugin block does.

**Comment by jhansche** [comment]
This can also be done by adding it to a different configuration (not `implementation`, which will influence the published dependencies):
```
val dummy by configurations.creating {
    isCanBeResolved = true
}
dependencies {
    // use whatever name was in settings.gradle; or "libs" for the default gradle/libs.versions.toml catalog
    versionCatalogs.named("publishedLibs").apply {
        libraryAliases.forEach { alias ->
            findLibrary(alias).ifPresent {
                addProvider(dummy.name, it)
            }
        }
        pluginAliases.forEach { alias ->
            findPlugin(alias).ifPresent {
                val mapped: Provider<Dependency> = it.map { plugin ->
                    // assumes that we can find the plugin using the .gradle.plugin marker:
                    create("${plugin.pluginId}:${plugin.pluginId}.gradle.plugin:${plugin.version.requiredVersion}")
                }
                addProvider(dummy.name, mapped)
            }
        }
    }
}
```
This allows the DependencyUpdatesTask to pick it up automatically and include the updates in that report.
